[Tan Yi Long] iP

yiiilonggg opened the pull request:
# Dumpling, A Tasty Chatbot
> To eat or not to eat

Like many dumplings, you can have all you like in one bite! Things like

- **Fast** CLI use
- _Reliable_ data management

People who might want to use this include 

1. Those who are **good** at CLI
2. Those who **enjoy** dumplings

You can run the App from the command line using: `java -jar ip`

Things to do
- [ ] manage dates better 📆 
- [ ] reminder system

View my profile [here](https://github.com/yiiilonggg)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yiiilonggg committed:
Complete Level 0.

Rename chatbot from Duke to Dumpling.
Implemented greet and exit.

yiiilonggg committed:
Completed Level 1.

Add echo functionality.

yiiilonggg committed:
Complete Level-2.

Shift main to new Main class.
Add Todo class to handle tasks.

yiiilonggg committed:
Complete Level-3.

Add mark and unmark functionality.
Add Task class.

yiiilonggg committed:
Complete Level-4.

Rename Todo to TaskList.
Add ToDo, Deadline, Event classes as subclasses of Task.

yiiilonggg committed:
Complete A-TextUiTesting.

Add Level-4 expected output and input as sample test case.

yiiilonggg committed:
Complete Level-5.

Add EmptyDescriptionException and InvalidCommandException.
Handle IOOBE and NumberFormatException.

yiiilonggg committed:
Complete Level-6.

Add delete functionality.
Create Pair class.
Update test case.

yiiilonggg committed:
Complete A-Enums.

Add Command enum.

yiiilonggg committed:
Complete Level-7

Task class: Convert to abstract class with abstract method getTaskForSaving
Deadline, Event, ToDo class: add getTaskForSaving method
InvalidTaskType exception: to handle corrupted data
TaskList class: add method to return string of data in saving format
Dumpling: add save and load data functionalities

yiiilonggg committed:
Add data folder into gitignore

yiiilonggg committed:
Merge branch 'branch-Level-7'

Add save and load data functionality

yiiilonggg committed:
Add dates to Deadline

yiiilonggg committed:
Merge branch 'branch-Level-8'

Add dates to Deadline

yiiilonggg committed:
Refactor code base

Remove all custom exceptions and replace with DumplingException.
DumplingException's message holds the custom information.
Catch different types of exceptions and throw a DumplingException instead.
Command enum class rename to CommandEnum.
Command class is an abstract base class.
{Add,Bye,Delete,List,Mark}Command inherit from Command.
Add Parser class to handle string parsing.
Add Storage class to maintain reading and writing to local file.
Add Ui class to handle printing to command line.
Refactor Dumpling and TaskList classes.

yiiilonggg committed:
Merge branch 'branch-A-MoreOOP'

yiiilonggg committed:
Refactor to organise code into packages

Command related classes in dumpling.command.
Task related classes in dumpling.task.
All other classes in dumpling.

yiiilonggg committed:
Merge branch 'branch-A-Packages'

yiiilonggg committed:
Merge branch 'branch-A-Gradle'

yiiilonggg committed:
Refactor backend load from Storage to Parser

yiiilonggg committed:
Add test cases for Parser

Test invalid date format for deadline classes.
Test missing date for deadline classes.
Test empty description for adding.
Methods tested: add, load

yiiilonggg committed:
Merge branch 'branch-A-JUnit'

yiiilonggg committed:
Add JavaDoc comments

yiiilonggg committed:
Fix coding style violations

yiiilonggg committed:
Add find functionality

Allow user to find tasks that have a partial match with the task description.

yiiilonggg committed:
Merge branch 'branch-A-JavaDoc'

yiiilonggg committed:
Merge branch 'branch-A-CodingStandard'

yiiilonggg committed:
Merge 'branch-Level-9' onto 'master'

Resolved merge conflicts.

bigismols commented on src/main/java/dumpling/Parser.java:
```diff
@@ -0,0 +1,217 @@
+                            stringCommand));
+        }
+        switch (commandEnum) {
+        case TODO:
```
I like how you formatted your switch statements, together with your enums! Many would forget the // fall through comments.

bigismols commented on src/main/java/dumpling/Ui.java:
```diff
@@ -0,0 +1,45 @@
+        this.echo(DIVIDER);
+    }
+
+    public void showWelcome() {
```
Missing JavaDoc comments for some public methods including this one! Minimally just a header comment should suffice.

bigismols commented on src/main/java/dumpling/command/Command.java:
```diff
@@ -0,0 +1,24 @@
+package dumpling.command;
```
Well done on abstracting wherever you could. I really like how all your classses are organized and abstracted into packages!

bigismols commented on src/main/java/dumpling/Storage.java:
```diff
@@ -0,0 +1,58 @@
+     * @param tasks TaskList of current state of Dumpling
+     * @throws DumplingException Thrown when there is an issue writing or creating the file
+     */
+    public void save(TaskList tasks) throws DumplingException {
```
Should this be a singular noun like taskList instead? Technically, it is a singular task list with a list of tasks.

bigismols reviewed (commented):
Overall, I found your code really easy to read. From the level of abstraction to following almost all the coding standards, it's really well written! :) Just a few coding violations, namely JavaDoc comments and some naming conventions.

yiiilonggg committed:
Add GUI for App

Obtained and adapted code from JavaFx tutorial.
Add new method to get command messages from command classes.

yiiilonggg committed:
Merge branch 'branch-Level-10'

yiiilonggg committed:
Add checkstyle

yiiilonggg committed:
Merge branch 'branch-A-CheckStyle'

yiiilonggg committed:
Add varargs to Ui echo method

yiiilonggg committed:
Merge branch 'branch-A-Varargs'

yiiilonggg committed:
Add support for building JAR with Gradle

yiiilonggg committed:
Add assertion for assumptions

Assertion in adding type to ensure only DEADLINE, EVENT and TODO commands are invoked.
Catch IOOB exception in delete method of TaskList.

yiiilonggg committed:
Refactor for code style improvements

Shifted logic of retrieving CommandEnum to CommandEnum enum.
Abstracted portions of parse and add methods of Parser class.

yiiilonggg committed:
Add streams

Streamed methods in TaskList that requires looping.

yiiilonggg committed:
Merge pull request #1 from yiiilonggg/branch-A-Assertions

Add assertion for assumptions

yiiilonggg committed:
Resolve merge conflicts

yiiilonggg committed:
Merge branch 'master' into branch-A-Streams

yiiilonggg committed:
Merge pull request #2 from yiiilonggg/branch-A-CodeQuality

Branch a code quality

yiiilonggg committed:
Merge branch 'master' into branch-A-Streams

yiiilonggg committed:
Merge pull request #3 from yiiilonggg/branch-A-Streams

Branch a streams

yiiilonggg committed:
Add CI

yiiilonggg committed:
Merge branch 'branch-A-CI'

yiiilonggg committed:
Add notes extension

Allow users to add notes to tasks.
Format of note command is: note <1-based task index> <notes>
The only restriction is that the notes cannot contain a newline character.

yiiilonggg committed:
Merge branch 'branch-BCD-Extension'

yiiilonggg committed:
Merge branch 'master' into branch-A-CI

yiiilonggg committed:
Update CI yml file

yiiilonggg committed:
Merge branch 'branch-A-CI'

yiiilonggg committed:
Fix checkstyle violations

yiiilonggg committed:
Fix legacy test issue

yiiilonggg committed:
Add resizing window functionality

Adapted from JavaFx tutorial level-5.

yiiilonggg committed:
Add extra ui to highlight error messages

Base code adapted from JavaFx tutorial.

yiiilonggg committed:
Add minor ui improvements

Flip text box for Dumpling.
Add padding in text box.

yiiilonggg committed:
Merge branch 'branch-A-BetterGui'

yiiilonggg committed:
Edit output from Dumpling for personality

yiiilonggg committed:
Merge branch 'branch-A-Personality'

yiiilonggg committed:
Add exception handling for IOOB

Delete and Note commands updated.
Extra spaces issue handled similarly as missing information.

yiiilonggg committed:
Merge branch 'branch-A-MoreErrorHandling'

yiiilonggg committed:
Add more tests

Testing CommandEnum retrieveCommandEnum.
Testing TaskList list, mark, updateTaskNotes.

yiiilonggg committed:
Merge branch 'branch-A-MoreTesting'

yiiilonggg committed:
Add ui image

Minor JavaDoc changes.

yiiilonggg committed:
Update readme file

yiiilonggg committed:
Merge branch 'branch-A-UserGuide'

yiiilonggg committed:
Add minor fixes

Use date object for event's from and to fields.
Update storing event's from and to fields.
Update Parser's method of loading events.
Update readme and ui.

yiiilonggg committed:
Add minor fix to readme

yiiilonggg committed:
Add minor fixes
